bodybank/character_generator/forms.py

```diff
@@ -1,8 +1,15 @@
-from django.forms import ModelForm
-from character_generator.models import Ego
+from django.forms import ModelForm, Form, ModelChoiceField
+from character_generator.models import Ego, Morph, Background, Career
 
 
 class EgoForm(ModelForm):
     class Meta:
         model = Ego
         fields = ["name", "cog", "inte", "ref", "sav", "som", "wil"]
+
+
+class CharacterSheetForm(Form):
```

Comment:
What about using a `ModelForm` here? Closely linked to the model, so you don't have to define all the fields. Then you can indeed override the queryset to limit the choices, example: https://stackoverflow.com/a/27914030

Reply:
```
class CharacterSheetForm(ModelForm):
    class Meta
	model = CharacterSheet
	fields = [...]
	
    def __init__(self, **kwargs):
        self.user = kwargs.pop('user')  # do before calling super()
        super().__init__(self, **kwargs)
        self.fields['ego'].queryset = Ego.objects.filter(user=self.user)
```

Then use the form like `CharacterSheetForm(request.POST, user=request.user)`